### PR TITLE
feat: Add username validation middleware

### PR DESCRIPTION
**Pull request by OkelleyDevelopment**
This middleware and tests will return the 401 codes for missing JSON entries. Assuming correct format, then it will validate the username does NOT exist.

If it does then a 409 code is returned with a message explaining.

### CODE REVIEW

File: src/api/middleware/validate_username.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    if (!req.body.roleId) {
+      res
+        .status(MISSING_CREDENTIAL)
+        .send({ status: MISSING_CREDENTIAL, message: "Missing roleId" });
+      return
+    }
+
+    let name_candidate = req.body.username;
```
Comment (AceMouty) [inline src/api/middleware/validate_username.ts]: potentially redundant variable that's already defined on line 10? If so remove

File: src/api/middleware/validate_username.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    await user.retrieveUser(name_candidate).then((r) => {
+      console.log(`The result is: ${r}`);
+
+      if (r != undefined) {
```
Comment (AceMouty) [inline src/api/middleware/validate_username.ts]: replace with descriptive variable
eg
```js
const userExists = r != undefined
if(userExists){
  
 }
 ```

File: src/api/middleware/validate_username.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    if (!req.body.roleId) {
+      res
+        .status(MISSING_CREDENTIAL)
+        .send({ status: MISSING_CREDENTIAL, message: "Missing roleId" });
+      return
+    }
+
+    let name_candidate = req.body.username;
+    await user.retrieveUser(name_candidate).then((r) => {
+      console.log(`The result is: ${r}`);
```
Comment (AceMouty) [inline src/api/middleware/validate_username.ts]: update `r` to be something more descriptive

 remove debug code

**Review by AceMouty — CHANGES_REQUESTED** [review]

File: src/api/middleware/validate_username.ts (hunk 1 above)
Comment (OkelleyDevelopment) [reply src/api/middleware/validate_username.ts]: Resolved in 

https://github.com/disney-parent-pass/back-end/pull/8/commits/e3847fde9f149cf3fdde0f77564e3c855f49da50

**Pull request merged**